### DISCUSSION TITLE
Custom Routes gui : screens for creating, viewing custom routes and writing js handler logic

### DISCUSSION THREAD

**Discussion by javea7171** [pocketbase/pocketbase#4870] [Q&A] [answered]
Essentially have a _custom_routes collection that enables you to view, create and edit routes inside of pocketbase. Collection fields contain http method , url scheme, middleware selections and code (logic) for that handler. Means that custom routes are instantly viewable within the application and if a basic code editor screen is implemented we can create and edit them directly. You could have a console at the foot of the code editor screen to get instant error feedback from the js runtime and also log to console for debugging.  All routing is then stored in the database directly and custom routing is better integrated into the application

**Comment by ganigeorgiev** [comment, marked as answer]
There are no plans for a "hooks editor" in the Admin UI for now if that is what you are asking (at least not out of the box).

Also note that currently routes and custom code in general cannot be loaded dynamically as it requires application restart. The primary way to extend PocketBase is with Go and this require compilation. The JSVM is just an optional "plugin".

**Reply by eqoram** [reply]
Will this change after the refactoring? Because I can imagine myself writing a Plugin to edit hooks from the Admin UI.
